0000-getting-started-cmake.md

```diff
@@ -0,0 +1,138 @@
+> 
+> This allows CMake builds to always get a well-defined environment for reproducible builds, making **build caching and remoting** possible.
+> 
+> Environment Descriptions are made of :
+>   - * A `CMAKE_TOOLCHAIN_FILE`, _e.g._ `environment/linux.cmake`
+>   - * An accompanying `.pkr.js` and `Dockerfile`, _e.g._ `environments/linux.pkr.js/`, `environments/linux.pkr.js/linux.Dockerfile`
+>
+> Multiple CMAKE_TOOLCHAIN_FILE with longer names than .pkr.js environments descriptions can share the same base system, more details in [Environments](/documentation/0400-environments).
+> 
+> Environments are not limited to docker containers, Virtual Machine images are also usable.
+> 
+> ```
+> ├── CMakeLists.txt
+> ├── environments
+> │   ├── linux.cmake
+> │   └── linux.pkr.js
+> │       ├── linux.Dockerfile
+> │       └── linux.pkr.js
+> └── test
+>     └── main.cpp
+> ```
+> 
+
+
+#### `CMakeLists.txt`
+```cmake
+cmake_minimum_required(VERSION 3.27.6)
+
+project(get-started)
+enable_testing()
+
+# Install fmt
+Include(FetchContent)
+set(FMT_TEST OFF CACHE BOOL "" FORCE)
+FetchContent_Declare(fmt
+  GIT_REPOSITORY https://github.com/fmtlib/fmt.git
+  GIT_TAG        9.0.0
+  OVERRIDE_FIND_PACKAGE
+)
+FetchContent_MakeAvailable(fmt)
+find_package(fmt REQUIRED)
+
+# Add a test executable
+add_executable(main test/main.cpp)
+target_link_libraries(main PRIVATE fmt::fmt)
+
+add_test(NAME main COMMAND ${CMAKE_CROSSCOMPILING_EMULATOR} $<TARGET_FILE:main> )
+```
+
+#### `test/main.cpp`
+```cpp
+#include <fmt/core.h>
+
+int main() {
+    fmt::print("Good Morning Caching & Hermetic Builds :-) \n"); 
+    return 0;
+}
+```
+
+#### `environments/linux.pkr.js/linux.pkr.js`
+```js
+{
+  "variables": { },
+  "builders": [
+    {
+      "type": "docker",
+      "image": "tipibuild/tipi-ubuntu:{{tipi_cli_version}}", // The Docker to use
+      "commit": true                                         // or update when Dockerfile present
+    }
+  ],
+  "post-processors": [
+    { 
+      "type": "docker-tag",
+      "repository": "linux",
+      "tag": "latest"
+    }
+  ],
+  "_tipi_version":"{{tipi_version_hash}}"
+}
+```
+Aside of the `CMAKE_TOOLCHAIN_FILE` the `pkr.js` folder specifies the environment that will be used for the build.
+This is a [packer docker builder configuration](https://developer.hashicorp.com/packer/integrations/hashicorp/docker/latest/components/builder/docker), **different to plain packer** `cmake-re` will build the Docker image if it cannot be found on a registry or if any change was made to the `linux.Dockerfile`.
+
+In this case we just take the default linux tipi environment with its clang toolchain. 
+
+More details in [Environments](/documentation/0400-environments)
+
+3. Cached, Reproducible Hermetic CMake builds :
+
+```bash
+$> cmake-re -S . -B build/ -DCMAKE_TOOLCHAIN_FILE=environments/linux.cmake
```

Comment:
Standardize on the same prompt everywhere? I think there's also a shell command further up in the page.

Reply:
Good point